fastlane_bot/events/pools/solidly_v2.py

```diff
@@ -120,11 +126,16 @@ async def async_update_from_contract(
         """
         reserve_balance = await contract.caller.getReserves()
 
-        try:
+        if hasattr(contract.caller, "factory"):
```

Comment:
This looks good.

Can we make it so the "else" statement can fail & add the ability to continue even if factory_address is None/empty string?

Reply:
1. The `else` statement will never end up setting `factory_address` to "None/empty string", as it reads the 256-bit value stored at slot 11, and then truncates it to a 160-bit address value. So for a non-cleopatra pool, it would just return some random address (whatever's stored at slot 11 of the given pool contract).

2. Even if it did return some "None/empty string", how exactly would you continue? The code further down the road needs the address of a valid factory contract, so it would just revert somewhere else along the way, as soon as it attempted to interact with that contract.

Reply:
Updated the `else` part such that if the `factory_address` is not the address of SOME contract (i.e., no bytecode deployed at that address), it is set to the zero-address.
Of course, there is not a huge gain in this, other than making potential problems easier to investigate.
First off, because we might end up with `factory_address` pointing to some non-factory contract, whose address happens to be stored in slot 11 of the pool contract.
Second, because a zero address will nevertheless throw an exception as soon as we attempt to interact with a factory contract "deployed" (i.e., obviously not deployed) at that address.
So again - we should probably conceive a different solution altogether for this entire feature of obtaining factory addresses (e.g., store them hard-coded somewhere).